contributing.md

```diff
@@ -0,0 +1,118 @@
+
+### Suggestions
+
+This section guides you through submitting an enhancement suggestion for Typedoc.
+
+Before creating a feature request, please check [this list][suggestions] to see if it has already been requested.
+
+When creating an enhancement request, explain your use case and ultimate goal. This will make it possible for contributors to suggest existing alternatives which may already meet your requirements.
+
+- **Use a clear and descriptive title** for the issue to identify the suggestion.
+- **Provide an example where this enhancement would improve TypeDoc**
+- **If possible, list another documentation generator where this feature exists**
+
+### Code
+
+Unsure of where to begin contributing to TypeDoc? You can start by looking through the issues labeled [good-first-issue] and [help-wanted]. Issues labeled with [good-first-issue] should only require changing a few lines of code and a test or two. Issues labeled with [help-wanted] can be considerably more involved and may require changing multiple files.
+
+For instructions on setting up your environment, see the [setup](#setup---git-github-and-node) instructions in this document.
+
+If you have started work on an issue and get stuck or want a second opinion on your implementation feel free to reach out through [Gitter].
+
+## Setup - Git, GitHub, and Node
+
+If you don't already have [Git] installed, install it first. You will need it to contribute to TypeDoc. You will also need to install [Node]. TypeDoc requires at least npm 4, so if you are running Node 7.3.0 or older you will need to upgrade npm using `npm install --global npm@^4`.
+
+#### Installation
+
+1. Fork the TypeDoc repository - https://github.com/TypeStrong/typedoc/fork
+1. Open a terminal, or "Git Bash" on Windows.
+1. Use `cd` to move to the directory that you want to work in.
+1. Clone your repository, replace USER with your GitHub username:
+   ```bash
+   git clone https://github.com/USER/typedoc
+   ```
+1. Add the TypeDoc repo as a remote repository
+   ```bash
+   git remote add typedoc https://github.com/TypeStrong/typedoc
+   ```
+1. Install dependencies and build the latest version:
+   ```bash
+   npm install
+   ```
+1. Open the typedoc folder in your favorite editor. If you don't have one, try [Visual Studio Code][VSCode] or [Atom]
+
+## Linting, Building, and Testing
+
+Once you have cloned TypeDoc, you can lint, build, and test the code from your terminal.
+
+#### Linting
+
+To lint the TypeDoc code, run `npm run lint`. This will start tslint and check all files for stylistic problems. You can also install a tslint plugin for your editor to show most style problems as you type.
+
+You can automatically fix some style problems by running `npm run lint -- --fix`.
+
+#### Building
+
+To compile the TypeDoc source, run `npm grunt`. This will start the TypeScript compiler and output the compiled JavaScript to the `dist` folder. If you want to build and test in one step, run `npm build`.
```

Comment:
Should be `npm run grunt` and `npm run build`?

Reply:
Fixed! That's what I get for using yarn...